src/control/server/instance_drpc_test.go

```diff
@@ -0,0 +1,170 @@
+
+	dc, err := instance.getDrpcClient()
+	if err != nil || dc == nil {
+		t.Fatal("Expected a dRPC client connection")
+	}
+
+	waitForIosrvReady(t, instance)
+}
+
+func TestIOServerInstance_CallDrpc(t *testing.T) {
+	for name, tc := range map[string]struct {
+		notReady bool
+		resp     *drpc.Response
+		expErr   error
+	}{
+		"not ready": {
+			notReady: true,
+			expErr:   errors.New("no dRPC client set"),
+		},
+		"success": {
+			resp: &drpc.Response{},
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			log, buf := logging.NewTestLogger(t.Name())
+			defer common.ShowBufferOnFailure(t, buf)
+			instance := getTestIOServerInstance(log)
+			if !tc.notReady {
+				cfg := &mockDrpcClientConfig{
+					SendMsgResponse: tc.resp,
+				}
+				instance.setDrpcClient(newMockDrpcClient(cfg))
+			}
+
+			_, err := instance.CallDrpc(drpc.ModuleMgmt, drpc.MethodPoolCreate, &mgmtpb.PoolCreateReq{})
+			common.CmpErr(t, tc.expErr, err)
+		})
+	}
+}
+
+func TestIOServerInstance_DrespToRankResult(t *testing.T) {
+	dRank := Rank(1)
+
+	for name, tc := range map[string]struct {
+		daosResp    *mgmtpb.DaosResp
+		inErr       error
+		targetState MemberState
+		junkRPC     bool
+		expResult   *MemberResult
+	}{
+		"rank success": {
+			expResult: &MemberResult{Rank: dRank, Action: "test", State: MemberStateJoined},
+		},
+		"rank failure": {
+			daosResp: &mgmtpb.DaosResp{Status: -1},
+			expResult: &MemberResult{
+				Rank: dRank, Action: "test", State: MemberStateErrored, Errored: true,
+				Msg: fmt.Sprintf("rank %d dRPC returned DER -1", dRank),
+			},
+		},
+		"drpc failure": {
+			inErr: errors.New("returned from CallDrpc"),
+			expResult: &MemberResult{
+				Rank: dRank, Action: "test", State: MemberStateErrored, Errored: true,
+				Msg: fmt.Sprintf("rank %d dRPC failed: returned from CallDrpc", dRank),
+			},
+		},
+		"unmarshal failure": {
+			junkRPC: true,
+			expResult: &MemberResult{
+				Rank: dRank, Action: "test", State: MemberStateErrored, Errored: true,
+				Msg: fmt.Sprintf("rank %d dRPC unmarshal failed: proto: mgmt.DaosResp: illegal tag 0 (wire type 0)", dRank),
+			},
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			_, buf := logging.NewTestLogger(t.Name())
+			defer common.ShowBufferOnFailure(t, buf)
+
+			if tc.daosResp == nil {
+				tc.daosResp = &mgmtpb.DaosResp{Status: 0}
+			}
+			if tc.targetState == MemberStateUnknown {
+				tc.targetState = MemberStateJoined
+			}
+
+			// convert input DaosResp to drpcResponse to test
+			rb := makeBadBytes(42)
+			if !tc.junkRPC {
+				rb, _ = proto.Marshal(tc.daosResp)
+			}
+			resp := &drpc.Response{
+				Status: drpc.Status_SUCCESS, // this will already have been validated by CallDrpc
+				Body:   rb,
+			}
+
+			gotResult := drespToMemberResult(Rank(dRank), "test", resp, tc.inErr, tc.targetState)
+			if diff := cmp.Diff(tc.expResult, gotResult, common.DefaultCmpOpts()...); diff != "" {
+				t.Fatalf("unexpected response (-want, +got)\n%s\n", diff)
+			}
+		})
+	}
+}
```

Comment:
Is it possible to add unit tests for the ping and prep shutdown calls? Not necessarily in this PR if they're complex.

Reply:
test coverage has been added elsewhere for the callers in this PR and will be added for these in the follow-on reformat PR

Reply:
these methods are covered by the `TestMgmtSvc_{Ping,PrepShutdown}Ranks` unit tests in `server/mgmt_system_test.go`